packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -151,13 +172,19 @@ class HeapTreeViewState extends State<HeapTree> with AutoDisposeMixin {
 
     addAutoDisposeListener(controller.selectedSnapshotNotifier, () {
       setState(() {
```

Comment:
not a new issue but it is odd we are calling setState around the calls to the controller. ideally setState should only be called if we are setting our own state not changing the controller. are there ValueListenables on the controller we could listen to from this view so we don't need the setState calls? That would make this more robust if we had multiple views displaying data from the controller at the same time.

Reply:
I'll review the code and fix this it's a good idea to be consistent in model/view.